runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -131,19 +133,47 @@ public void onChatMessage(ChatMessage event)
 		clue = null;
 	}
 
-	@Schedule(
-		period = 600,
-		unit = ChronoUnit.MILLIS
-	)
-	public void checkForClues()
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
 	{
-		if (client.getGameState() == GameState.LOGIN_SCREEN)
+		// Check if item was removed from inventory
+		if (clue != null && event.getItemContainer() == client.getItemContainer(InventoryID.INVENTORY))
 		{
-			client.clearHintArrow();
-			clue = null;
-			return;
+			final Item[] items = event.getItemContainer().getItems();
+			boolean found = false;
+
+			// Clue was maybe removed from inventory, check if there is any clue scrolls left
+			for (Item item : items)
+			{
+				final ItemComposition itemContainerDefinition = client.getItemDefinition(item.getId());
+
+				// Check if we have any clue scrolls or map clues left
+				if (MapClue.forItemId(item.getId()) != null || itemContainerDefinition.getName().startsWith("Clue scroll"))
```

Comment:
Why are map clues separate here? It looks like their names all begin with Clue scroll.

Reply:
Oh do they? I was not sure so I added special check.